### PR TITLE
Fix for persistent Horizontal pod autoscaling failures

### PR DESCRIPTION
**Pull request by dims**
We are waiting in a loop until timeout occurs by design in this test case.  `PollUntilContextTimeout will terminate polling after timeout duration by setting a context timeout.` note the AFTER in this sentence. However the context can get cancelled somewhere else as well. When this happens `cs.AppsV1().Deployments(namespace).Get` returns `context.DeadlineExceeded` which is by design NOT a problem. we were waiting for the timeout and it did, So instead of failing hard with `framework.Failf` we should return the err we got. If you see later in the method `ensureDesiredReplicasInRange` we do check for `wait.Interrupted` which includes `context.DeadlineExceeded`, so the test passes.

this takes care of some of the flakiness we see in the 2 tests we see here:
https://storage.googleapis.com/k8s-triage/index.html?pr=1&test=CustomMetricsAutoscaling#3ef126d1b789fe348a1d

Related to: https://github.com/kubernetes/kubernetes/issues/121738

#### What type of PR is this?
/kind cleanup
/kind failing-test

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by dims** [comment]
pull-kubernetes-e2e-autoscaling-hpa-cpu

**Comment by dims** [comment]
`pull-kubernetes-e2e-gci-gce-autoscaling` is not related

**Review by michelle192837 — APPROVED** [review]

File: test/e2e/autoscaling/custom_metrics_stackdriver_autoscaling.go [hunk 1]
```diff
@@ -609,11 +610,18 @@ func waitForReplicas(ctx context.Context, deploymentName, namespace string, cs c
...
+
 func ensureDesiredReplicasInRange(ctx context.Context, deploymentName, namespace string, cs clientset.Interface, minDesiredReplicas, maxDesiredReplicas int, timeout time.Duration) {
 	interval := 60 * time.Second
 	err := wait.PollUntilContextTimeout(ctx, interval, timeout, true, func(ctx context.Context) (bool, error) {
 		deployment, err := cs.AppsV1().Deployments(namespace).Get(ctx, deploymentName, metav1.GetOptions{})
 		if err != nil {
+			if interrupted(err) {
+				return true, err
+			}
 			framework.Failf("Failed to get replication controller %s: %v", deployment, err)
```
Comment (aojea) [inline test/e2e/autoscaling/custom_metrics_stackdriver_autoscaling.go:625]: the problem us here, we should not use a failf inside the loop and just return the error

**Review by pacoxu — COMMENTED** [review]
/lgtm

**Pull request merged**